### ISSUE TITLE
[Feature Request] Allow workers to pick up ioredis prefixes for ACL usage 

### ISSUE THREAD

**Issue by lukepolo** [taskforcesh/bullmq#915]
We have a lot of IoT devices that we would like to allow our workers to pick up based on a prefix. This would allow us to use ACL's to make sure they can only write / read from that prefix. 

Example : a edge device places a job onto the queue with `bullmq:queue:10.0.0.1:...`

The ACL would allow the edge to write it based on the prefix.  We currently are writing a LUA a lua script to handle this but its a pain.

**Comment by manast** [comment]
I am not sure I understand the use case correctly.  What would "pick up" mean in this context? Like you, you add jobs to a queue with a some prefix (could this be a group Id?) and then the workers what should they do different based on that prefix?

**Comment by lukepolo** [comment]
Groups may solve this because of how they are prefixed. 

is the key prefixed like 

`bullmq-prefix:queue-name:group-id:uniqueue-jobid` ?

**Comment by manast** [comment]
No. Jobs that are part of a group have the same key as standard jobs, the difference is that there is group list for every group that has jobs in it. The group list has this format: ```bull:queueName:groups:groupId```

**Comment by manast** [comment]
When the job is processed you can access the group Id, why can't you use that information to decide the ACL for the device?

**Comment by lukepolo** [comment]
these are edge devices so we dont want them to have access to anything but certain keys (redis ACL with prefixes)

This makes it so an edge device cannot get another devices information at the main redis instance 

We could've done this differently but we made the bed long time ago.

**Comment by manast** [comment]
@lukepolo you are welcome to discuss your requirements at more length here: https://github.com/taskforcesh/bullmq-pro-support/discussions or hit me at manast@taskforce.sh
